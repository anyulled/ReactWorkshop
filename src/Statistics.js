import React from "react";
import PropTypes from "prop-types";
import {ListGroup, ListGroupItem} from "react-bootstrap";

class Statistics extends React.Component {
    render() {
        const {props: {rainyDays, sunnyDays, loading, error}} = this;
        return (!loading && !error && <ListGroup>
            <ListGroupItem bsStyle="warning"><strong>Sunny Days:</strong> {sunnyDays}</ListGroupItem>
            <ListGroupItem bsStyle="info"><strong>Rainy Days:</strong> {rainyDays}</ListGroupItem>
        </ListGroup>);
    }
}
Statistics.propTypes = {
    rainyDays: PropTypes.number,
    sunnyDays: PropTypes.number
};

Statistics.defaultProps = {
    rainyDays: 0,
    sunnyDays: 0
};

export default Statistics;