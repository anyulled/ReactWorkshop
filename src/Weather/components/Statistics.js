import React from "react";
import PropTypes from "prop-types";
import {ListGroup, ListGroupItem} from "react-bootstrap";

class Statistics extends React.Component {
    render() {
        const {props: {rainyDays, sunnyDays, loading, error}} = this;
        return (!loading && !error && <ListGroup>
            <ListGroupItem bsStyle="warning"><strong>Sunny Days:</strong> {sunnyDays}</ListGroupItem>
            <ListGroupItem bsStyle="info"><strong>Rainy Days:</strong> {rainyDays}</ListGroupItem>
        </ListGroup>);
    }
}
Statistics.propTypes = {
    error: PropTypes.bool,
    loading: PropTypes.bool,
    rainyDays: PropTypes.number,
    sunnyDays: PropTypes.number
};

Statistics.defaultProps = {
    error: false,
    loading: false,
    rainyDays: 0,
    sunnyDays: 0
};

export default Statistics;