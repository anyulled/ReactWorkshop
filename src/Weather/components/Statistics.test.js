import React from "react";
import Statistics from "./Statistics";
import {shallow} from "enzyme";

describe("Statistics", () => {
    it("Renders properly", () => {
        const wrapper = shallow(<Statistics/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("should display sunny and rainy days correctly", () => {
        const props = {
            sunnyDays: 50,
            rainyDays: 10
        };
        const wrapper = shallow(<Statistics {...props}/>);
        expect(wrapper.find("ListGroupItem").at(0).html()).toContain(`<strong>Sunny Days:</strong> ${props.sunnyDays}</span>`);
        expect(wrapper.find("ListGroupItem").at(1).html()).toContain(`<strong>Rainy Days:</strong> ${props.rainyDays}</span>`);
    });
});