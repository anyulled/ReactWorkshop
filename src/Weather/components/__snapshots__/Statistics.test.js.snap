// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics Renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Statistics
    error={false}
    loading={false}
    rainyDays={0}
    sunnyDays={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "bsClass": "list-group",
      "children": Array [
        <ListGroupItem
          bsClass="list-group-item"
          bsStyle="warning"
          listItem={false}
        >
          <strong>
            Sunny Days:
          </strong>
           
          0
        </ListGroupItem>,
        <ListGroupItem
          bsClass="list-group-item"
          bsStyle="info"
          listItem={false}
        >
          <strong>
            Rainy Days:
          </strong>
           
          0
        </ListGroupItem>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "bsClass": "list-group-item",
          "bsStyle": "warning",
          "children": Array [
            <strong>
              Sunny Days:
            </strong>,
            " ",
            0,
          ],
          "listItem": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Sunny Days:",
            },
            "ref": null,
            "rendered": "Sunny Days:",
            "type": "strong",
          },
          " ",
          0,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "bsClass": "list-group-item",
          "bsStyle": "info",
          "children": Array [
            <strong>
              Rainy Days:
            </strong>,
            " ",
            0,
          ],
          "listItem": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Rainy Days:",
            },
            "ref": null,
            "rendered": "Rainy Days:",
            "type": "strong",
          },
          " ",
          0,
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "bsClass": "list-group",
        "children": Array [
          <ListGroupItem
            bsClass="list-group-item"
            bsStyle="warning"
            listItem={false}
          >
            <strong>
              Sunny Days:
            </strong>
             
            0
          </ListGroupItem>,
          <ListGroupItem
            bsClass="list-group-item"
            bsStyle="info"
            listItem={false}
          >
            <strong>
              Rainy Days:
            </strong>
             
            0
          </ListGroupItem>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "bsClass": "list-group-item",
            "bsStyle": "warning",
            "children": Array [
              <strong>
                Sunny Days:
              </strong>,
              " ",
              0,
            ],
            "listItem": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Sunny Days:",
              },
              "ref": null,
              "rendered": "Sunny Days:",
              "type": "strong",
            },
            " ",
            0,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "bsClass": "list-group-item",
            "bsStyle": "info",
            "children": Array [
              <strong>
                Rainy Days:
              </strong>,
              " ",
              0,
            ],
            "listItem": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Rainy Days:",
              },
              "ref": null,
              "rendered": "Rainy Days:",
              "type": "strong",
            },
            " ",
            0,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
